src/Orleans.EventSourcing/JournaledGrain.cs

```diff
@@ -56,60 +56,73 @@ protected JournaledGrain(IGrainIdentity identity, IGrainRuntime runtime)
         }
 
         /// <summary>
-        /// Raise an event.
+        /// Raises an event, and either confirms it, or throws <see cref="LostRaceException"/> 
+        /// and discards it (if another instance raised an event in the meantime).
         /// </summary>
         /// <param name="event">Event to raise</param>
-        /// <returns></returns>
-        protected virtual void RaiseEvent<TEvent>(TEvent @event) 
+        /// <returns>a task that completes when the event has been confirmed</returns>
+        protected async Task RaiseEvent<TEvent>(TEvent @event) 
```

Comment:
Migrations of old code will be error-prone when this method is not awaited. Maybe it is better to rename the method, so code breaks at compile time instead of runtime.

Reply:
Good point. I think for this particular method, we can just use `RaiseEventAsync`, which is a better name anyway.